examples/riot_and_cpp/main.cpp

```diff
@@ -24,11 +24,9 @@
 /*
  * all included headers defining c functions, i.e. all RIOT functions, must be marked as extern "C"
  */
-extern "C" {
```

Comment:
Maybe update the comment above (or simply remove it ?)

Reply:
This is actually intended as an example, therefore I suggest to keep the extern C, but the RIOT include can move outside of it, because all RIOT headers have extern C in them.

Reply:
the `c_functions.h` (which is part of this example) also contains an `extern "C"` so this is really superfluous here. ;-)

Reply:
Yes, I understand that, but from a pedagogical perspective it might make sense to leave the extern C in place. But whatever, the person using this example should not expect to learn C++ from here, there are much better resources for that.

Reply:
I think what @gebart suggested would be the right move to help newcomers ...

> This is actually intended as an example, therefore I suggest to keep the extern C, but the RIOT include can move outside of it, because all RIOT headers have extern C in them.

... and remove the `extern "C"` in the example header ([`c_functions.h`](https://github.com/RIOT-OS/RIOT/blob/master/examples/riot_and_cpp/c_functions.h)), move the riot header outside the block and add a comment to explain the difference. Something like this?

```cpp
/*
 * Including C from C++ usually requires an extern C block around the code.
 * Since RIOT aims to make C++ programming easy, all internal headers
 * already include the block and can be included without problems.
 */
```

Alternatively, if you consider this general knowledge, remove the `extern "C"` block and the comment.

Reply:
I think, that if the RIOT claims itself to be c++ compatible, than it must have `extern "C"` in all headers. So I prefer the "general knowledge" solution.

Reply:
The header in question (`c_functions.h`) is not a RIOT "internal" header, but shipped with the example. As such it could be helpful to have the extern in the main to give newcomers a hint on how to include headers from external sources.

Then again I'm not set on this. If you think this is better (and @gebart doesn't seem to mind as well), we can remove the extern C here. The comment should be removed or updated accordingly.